src/flamegpu/gpu/CUDAEnsemble.cu

```diff
@@ -224,6 +226,29 @@ unsigned int CUDAEnsemble::simulate(const RunPlanVector &plans) {
         printf("Ensemble time elapsed: %fs\n", ensemble_elapsed_time);
     }
 
+    // Send Telemetry
+    if (config.telemetry) {
+        // Generate some payload items
+        std::map<std::string, std::string> payload_items;
+        payload_items["GPUDevices"] = flamegpu::util::detail::compute_capability::getDeviceNames(config.devices);
```

Comment:
With regards to (b), you could build the set with the device ids included in `devices`. This set is built earlier in the method, based on the users configuration.

Reply:
@Robadob it takes the std::set of devices as an argument from teh ensemble config so it should be just the devices used (not actually got to that bit of code yet nor tested this)

Reply:
Actual device properties would be useful too / instead of name if it will actually guide anything about which devices to support (i.e. major cuda arch would mean we don't have to manually look up the compute capability for each named device)

Reply:
Fair I missed it was being passed `config.devices`. (Though I haven't looked at the difference between how `CUDAEnsemble` and that util method filter for invalid device IDS, or empty set.